tuf/signed/verify.go

```diff
@@ -105,3 +106,31 @@ func VerifySignature(msg []byte, sig data.Signature, pk data.PublicKey) error {
 	}
 	return nil
 }
+
+// VerifyPublicKeyMatchesPrivateKey checks that the specified private key and public key together form a valid keypair.
+func VerifyPublicKeyMatchesPrivateKey(privKey data.PrivateKey, pubKey data.PublicKey) error {
```

Comment:
Rather than having to sign and verify, I think we should be able to verify that the keys match via the `data.PrivateKey.Public()` function (or by comparing the modulus of the public key on the cert)